rcl/src/rcl/node.c

```diff
@@ -72,18 +79,46 @@ rcl_node_init(rcl_node_t * node, const char * name, const rcl_node_options_t * o
     allocator->allocate, "allocate not set", return RCL_RET_INVALID_ARGUMENT);
   RCL_CHECK_FOR_NULL_WITH_MSG(
     allocator->deallocate, "deallocate not set", return RCL_RET_INVALID_ARGUMENT);
+  // Make sure the node name is valid before allocating memory.
+  int validation_result = 0;
+  size_t invalid_index = 0;
+  ret = rmw_validate_node_name(name, &validation_result, &invalid_index);
+  if (ret != RMW_RET_OK) {
+    RCL_SET_ERROR_MSG(rmw_get_error_string_safe());
+    return ret;
+  }
+  if (validation_result != RMW_NODE_NAME_VALID) {
+    const char * msg = rmw_node_name_validation_result_string(validation_result);
```

Comment:
i think this was discussed in another PR already. But why not apply the same logic as above here?
```
// sets the error message inside
ret = rmw_node_name_validation_result_string(validation_result);
if (ret != RMW_RET_OK)  // or similar for this use-case
{
  RCL_GET_ERROR_MSG_SAFE...
}
```